### PR TITLE
[ISSUE-647] fix event trigger, change zip action

### LINKED ISSUE #685: E2E workflow trigger
E2E workflow triggered by any comment in pull requests or issues

Related code https://github.com/dell/csi-baremetal/blob/master/.github/workflows/e2e.yml#L20

### PR DESCRIPTION
**Pull request by rkiselenko**
### Resolves #685

Change zip action, fix trigger event

## PR checklist
- [x] Add link to the issue
- [x] Choose Project
- [x] Choose PR label
- [ ] New unit tests added
- [ ] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved

## Testing
_Provide test details_

### CODE REVIEW

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -207,11 +207,9 @@ jobs:
...
 
-    - name: Archive logs
-      uses: papeloto/action-zip@v1
+    - uses: montudor/action-zip@v1
       with:
-        files: test/e2e/reports
-        dest: reports.zip
+        args: zip -qq -r reports.zip test/e2e/reports
 
     - name: Upload report to artifacts
```
Comment (safronovD) [inline .github/workflows/e2e.yml:214]: What happens, if we try to upload a folder here?

Comment (rkiselenko) [reply .github/workflows/e2e.yml:214]: https://github.com/actions/upload-artifact#zipped-artifact-downloads

Comment (safronovD) [reply .github/workflows/e2e.yml:214]: thank you

**Review by safronovD — APPROVED** [review]

File: .github/workflows/e2e.yml [hunk 2]
```diff
@@ -17,7 +17,7 @@ env:
 jobs:
   e2e:
     # we need to start validation on master push or on /ci comment in PR only
-    if: github.event.issue.pull_request == false || (github.event.issue.pull_request && startsWith(github.event.comment.body, '/ci'))
+    if: github.event_name == 'push' || (github.event_name == 'issue_comment' && startsWith(github.event.comment.body, '/ci'))
```
Comment (mishoyama) [inline .github/workflows/e2e.yml:20]: Why do you use `issue_comment` here?
```
issue_comment - Activity related to an issue or pull request comment.
```
We don't want to run this action when someone posted comment in issue

**Review by mishoyama — CHANGES_REQUESTED** [review]

File: .github/workflows/e2e.yml (hunk 2 above)
Comment (rkiselenko) [reply .github/workflows/e2e.yml:20]: it's an event name https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#issue_comment

Comment (rkiselenko) [reply .github/workflows/e2e.yml]: check for pull request [added](https://github.com/dell/csi-baremetal/pull/687/commits/67fea6763931b1cab188afbdbdc21dfc51a8dae5)

**Review by yurkov-anton — APPROVED** [review]

**Review by mishoyama — APPROVED** [review]

**Pull request merged**